data/codings.go

```diff
@@ -49,25 +49,52 @@ func decode(data []byte, decoder *encoding.Decoder) (st string, err error) {
 	return
 }
 
+type CustomEncoding struct {
+	encDec EncDec
+	coding byte
+}
+
+func NewCustomEncoding(coding byte, encDec EncDec) Encoding {
+	return &CustomEncoding{
+		coding: coding,
+		encDec: encDec,
+	}
+}
+
+// Encode string.
+func (c *CustomEncoding) Encode(str string) ([]byte, error) {
```

Comment:
`c` has no need to be a pointer here and I think it's the same for all bellow

Reply:
@goten4 the reason to use pointer for struct func is: prevent copy